components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/AccessTokenIssuer.java

```diff
@@ -305,7 +310,21 @@ public OAuth2AccessTokenRespDTO issue(OAuth2AccessTokenReqDTO tokenReqDTO)
 
         if (tokReqMsgCtx.getScope() != null && OAuth2Util.isOIDCAuthzRequest(tokReqMsgCtx.getScope())) {
             IDTokenBuilder builder = OAuthServerConfiguration.getInstance().getOpenIDConnectIDTokenBuilder();
-            tokenRespDTO.setIDToken(builder.buildIDToken(tokReqMsgCtx, tokenRespDTO));
+            String idToken = builder.buildIDToken(tokReqMsgCtx, tokenRespDTO);
+            try {
+                JWT jwt = JWTParser.parse(idToken);
```

Comment:
@kasunbg 
If the token is invalid here I return tokenRespDTO = handleError(
                        OAuth2ErrorCodes.SERVER_ERROR,
                        "Server Error",
                        request.getOauth2AccessTokenReqDTO());
Currently in DefaultIDTokenBuilder, it returns PlainJWT[1] or signJWT[2]. So, If I do the validation for the token there, I need to return some String to indicate the invalid token.
Shall I send some error string or 'tokenRespDTO.getErrorMsg()' ?

Appreciate some suggestion on this.

[1]https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/6855d9a8bd26c74b2cdd6bdf02ccd2be7ba75eea/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/DefaultIDTokenBuilder.java#L316
[2]https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/6855d9a8bd26c74b2cdd6bdf02ccd2be7ba75eea/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/DefaultIDTokenBuilder.java#L318